src/config.rs

```diff
@@ -211,22 +212,32 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
         Some(parse_default(CA_ADDRESS, default_istiod_address)?)
     });
 
-    let xds_root_cert_provider = parse_default(
-        XDS_ROOT_CERT_PROVIDER_ENV,
-        DEFAULT_ROOT_CERT_PROVIDER.to_string(),
-    )?;
+    let xds_root_cert_provider =
+        parse_default(XDS_ROOT_CA_ENV, DEFAULT_ROOT_CERT_PROVIDER.to_string())?;
     let xds_root_cert = if Path::new(&xds_root_cert_provider).exists() {
-        RootCert::File(xds_root_cert_provider.parse().unwrap())
+        let xds_root_cert_match = match xds_root_cert_provider.parse::<PathBuf>() {
+            Ok(root_cert) => root_cert,
+            Err(_) => PathBuf::new(),
+        };
+        RootCert::File(xds_root_cert_match)
```

Comment:
We need to actually handle the error

```suggestion
        RootCert::File(xds_root_cert_provider.parse()?)
```

Reply:
Hi @howardjohn, can not directly use `?` to handle such error, I encountered below error message:

```
expected enum `Infallible`, found enum `config::Error`
```

so I use `match` statement to handle the Err situation.

Reply:
Its not handled, the error is ignored and we just return an invalid PathBuf::new()... we need to actually return an error.

` RootCert::File(xds_root_cert_provider.into())` will do it tohugh

Reply:
done!